lib/src/widgets/nes_button.dart

```diff
@@ -54,7 +54,10 @@ class NesButton extends StatefulWidget {
   final Widget child;
 
   /// The callback called the button is pressed.
-  final VoidCallback onPressed;
+  final VoidCallback? onPressed;
+
+  /// Whether the button is disabled.
+  final bool isDisabled;
```

Comment:
Since this variable is an internal thing and calculated automatically, can we make it private?

Reply:
yes, sure! I will also provide some tests

Reply:
@erickzanardo done, the tests are updated. I also added some NesButtonType golden test, I prefer them over these simple widget test. Because you can check if the widget it self renders correctly and also the color is correct.